src/todo.ts

```diff
@@ -0,0 +1,99 @@
+import { Item } from './item';
+class TodoApp {
+    /**
+     * Allows the creation and maintenance of a simple todo list
+     */
+
+    public toDoList: Map<number, Item>;
+    public curId: number;
+
+    constructor() {
+        this.curId = 1;
+        this.toDoList = new Map<number, Item>();
+    }
+
+    // Had to generate the item ID here so two todoApps could have separate id lists
+    public addItem(description: string, dueAt: string): Item {
+        const date = new Date(dueAt);
+        if (date.toString() === 'Invalid Date') {
+            throw new Error('The date you entered is invalid');
+        }
+        const item = new Item(description, date, this.curId++);
+        this.toDoList.set(item.id, item);
+        return item;
+    }
+    public checkItem(id: number): boolean {
+        if (!this.toDoList.has(id)) {
+            throw new Error('No such item exists');
+        }
+        this.toDoList.get(id).done = true;
+        return true;
+    }
+    public uncheckItem(id: number): boolean {
+        if (!this.toDoList.has(id)) {
+            throw new Error('No such item exists');
+        }
+        this.toDoList.get(id).done = false;
+        return true;
+    }
+    public removeItem(id: number): Item {
+        if (!this.toDoList.has(id)) {
+            throw new Error('No such item exists');
+        }
+        const deleted = this.toDoList.get(id);
+        this.toDoList.delete(id);
+        return deleted;
+    }
+
+    public listAllToDos(): Item[] {
+        const ids = Array.from(this.toDoList.values());
+        return ids;
+    }
+
+    public listCompletedToDos(): Array<Array<string | number | Date | boolean>> {
+        if (!this.toDoList.size) {
```

Comment:
Why is this an error?  Shouldn't it just return an empty list?

Reply:
Applies throughout.  This doesn't seem like an error condition.  Rather just a list with no results. Am I wrong?